pkg/controllers/nodeclaim/tagging/controller.go

```diff
@@ -120,7 +125,9 @@ func (c *Controller) tagInstance(ctx context.Context, nc *karpv1.NodeClaim, id s
 
 func isTaggable(nc *karpv1.NodeClaim) bool {
 	// Instance has already been tagged
-	if val := nc.Annotations[v1.AnnotationInstanceTagged]; val == "true" {
+	instanceTagged := nc.Annotations[v1.AnnotationInstanceTagged]
+	clusterNameTagged := nc.Annotations[v1.AnnotationClusterNameTaggedCompatability]
+	if instanceTagged == "true" && clusterNameTagged == "true" {
```

Comment:
If only one of these is tagged, do we idempotently add the one that's already in or does that fail?

Reply:
> do we idempotently add the one that's already in or does that fail

It doesn't fail, we would just only add the tags that aren't there on the resource and not update the others